src/client.ts

```diff
@@ -9391,7 +9391,10 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
                 ((<MatrixError>err).httpStatus === 400 ||
                     // This the correct standard status code for an unsupported
                     // endpoint according to MSC3743.
-                    (<MatrixError>err).httpStatus === 404)
+                    (<MatrixError>err).httpStatus === 404 ||
+                    // This the correct standard status code for an invalid
+                    // method according to MSC3743.
```

Comment:
both error codes are the correct standard status codes - comment should be moved up a line and adjusted

Reply:
Fixed in c7c16256d